lib/index.js

```diff
@@ -1,343 +1,381 @@
+                                );
+                            });
 
-                    return '';
-                });
+                        const compiler = Handlebars.compile(String(Fs.readFileSync(__dirname + '/hbs/migrationFileTemplate.hbs')));
+                        const createdTables = tableDeltas.tableDelta.create.concat(tableDeltas.joinTableDelta.create);
 
-                Fs.readdirSync(`${__dirname}/hbs/partials/`)
-                    .forEach((fileName) => {
+                        // Need this message for both cases where no migration is needed
+                        // (because unsupported columns get skipped)
+                        // and on a successful migration file generation where other
+                        // columns are changed or tables added, etc.
+                        let skippedColsMsg = '';
+                        if (!!internals.skippedCols.length) {
+                            const dedupedSkippedCols = internals.skippedCols.filter((skippedInfo, i) => {
 
-                        Handlebars.registerPartial(
-                            fileName.split('.hbs')[0],
-                            String(Fs.readFileSync(`${__dirname}/hbs/partials/${fileName}`))
-                        );
-                    });
+                                return internals.skippedCols.indexOf(skippedInfo) === i;
+                            });
+                            skippedColsMsg = `Skipped unsupported columns:${Os.EOL}${dedupedSkippedCols.join(Os.EOL)}`;
+                        }
 
-                const compiler = Handlebars.compile(String(Fs.readFileSync(__dirname + '/hbs/migrationFileTemplate.hbs')));
-                const createdTables = tableDeltas.tableDelta.create.concat(tableDeltas.joinTableDelta.create);
+                        if (createdTables.length === 0 &&
+                            columnDeltas.changes.length === 0) {
 
-                if (createdTables.length === 0 &&
-                    columnDeltas.changes.length === 0) {
+                            const noMigrationMsg = !!skippedColsMsg ? `${skippedColsMsg}${Os.EOL}No migration needed` : 'No migration needed';
+                            return cb(null, noMigrationMsg);
+                        }
 
-                    return cb(null, 'No migration needed');
-                }
+                        const migrationFileContents = compiler({
+                            allTables,
+                            createdTables,
+                            tableDeltas,
+                            columnDeltas
+                        });
 
-                const migrationFileContents = compiler({
-                    allTables,
-                    createdTables,
-                    tableDeltas,
-                    columnDeltas
-                });
+                        Promise.resolve(migrationGroup.knex.migrate.make(userOptions.migrationName || 'schwifty-migration', {
```

Comment:
Huh, I would expect `migrationGroup.knex.migrate.make()` to already be a bluebird promise!  Can we ensure it isn't?

Reply:
I was having issues when modifying knexes for tests where this promise lost the `.asCallback` ability -- super weird! I will follow this note and your other one from above and re-check all of these for already having `.asCallback`

Reply:
Ok -- looking back -- I think I'll have to get cleverer with modifying the knex queryBuilder select func

Reply:
_Ohh_ you mean in the tests?

Reply:
Yes in the tests sorry